core/sql/common/BaseTypes.cpp

```diff
@@ -412,6 +419,13 @@ short convertTypeToText_basic(char * text,	   // OUTPUT
 	str_sprintf(text, "NUMERIC(%d, %d) UNSIGNED", precision, scale);
       break;
 
+    case REC_BIN8_UNSIGNED:
+      if (!precision)
+        str_sprintf(text, "TINYINT UNSIGNED");
+      else
+	str_sprintf(text, "NUMERIC(%d, %d)", precision, scale);
```

Comment:
At current time, NUMERIC datatypes are handled as smallint, int or largeint.
They are not converted to tinyint. That could be done at some point.
So this code will not be reached.

But review comment is correct about this code being incorrect and I have fixed it.
If we do enhance NUMERIC with precision <= 2 to be treated as tinyint, then
this code will be exercised.